internal/logger/logger.go

```diff
@@ -114,26 +136,37 @@ func (f *loggerFactory) newLogger(level, prefix string) *log.Logger {
 
 func (f *loggerFactory) writer(level string) io.Writer {
 	if f.file != nil {
-		switch f.format {
-		case "json":
+		if f.format == "json" {
 			return &jsonWriter{
 				w:     f.file,
 				level: level,
 			}
-		case "text":
+		} else {
 			return &textWriter{
 				w:     f.file,
 				level: level,
 			}
 		}
-	}
-
-	switch level {
-	case "NOTICE":
-		return daemonize.StatusWriter
-	case "ERROR":
-		return os.Stderr
-	default:
-		return os.Stdout
+	} else if f.sysWriter != nil {
+		if f.format == "json" {
+			return &jsonWriter{
+				w:     f.sysWriter,
+				level: level,
+			}
+		} else {
+			return &textWriter{
+				w:     f.sysWriter,
+				level: level,
+			}
```

Comment:
please extract this logic into private method and use it for both file and syswriter